src/shared/watchdog.c

```diff
@@ -17,11 +17,55 @@
 static int watchdog_fd = -1;
 static char *watchdog_device = NULL;
 static usec_t watchdog_timeout = USEC_INFINITY;
+static usec_t watchdog_pretimeout = USEC_INFINITY;
 static usec_t watchdog_last_ping = USEC_INFINITY;
 
-static unsigned int usec_to_sec(usec_t val) {
+/* Starting from kernel version 4.5, the maximum allowable watchdog timeout is
+ * UINT_MAX/1000U seconds (since internal calculations are done in milliseconds).
+ */
+#define WATCHDOG_TIMEOUT_MAX_SEC (UINT_MAX/1000U)
+
+static unsigned usec_to_sec(usec_t val) {
```

Comment:
maybe name this saturated_usec_to_sec(), since the saturation is kinda what it is about